Src/ILGPU/Runtime/Cuda/CudaStream.cs

```diff
@@ -83,13 +91,16 @@ public override void Synchronize()
         /// <summary cref="DisposeBase.Dispose(bool)"/>
         protected override void Dispose(bool disposing)
         {
-            if (streamPtr != IntPtr.Zero)
+            if (responsibleForHandle)
             {
-                CudaException.ThrowIfFailed(
-                    CudaAPI.Current.DestroyStream(streamPtr));
-                streamPtr = IntPtr.Zero;
+                if (streamPtr != IntPtr.Zero)
+                {
+                    CudaException.ThrowIfFailed(
+                        CudaAPI.Current.DestroyStream(streamPtr));
+                    streamPtr = IntPtr.Zero;
+                }
+                base.Dispose(disposing);
```

Comment:
ISSUE: We should always call `base.Dispose(disposing);`, even when we are not responsible for disposing the stream.

This is because the base class still needs to cleanup after itself.